### ISSUE TITLE
Bugs in updated logger

### ISSUE THREAD

**Issue by jakehader** [terrapower/armi#366]
Recently, #336 was merged in and I have noticed several bugs that are now present when running cases:

1. All outputs are being sent to the standard error (stderr) stream.
2. Some MPI messages appear to be missing from the outputs, even for the head process (MPI_RANK=0)
3. I received the following logger error after a run was "completed" causing the case to fail:

```
[CRIT 004 ] Stack trace: Traceback (most recent call last):
  File "...\home\jhader\natrium\framework\armi\operators\operatorMPI.py", line 103, in operate
    self.workerOperate()
  File "...\home\jhader\natrium\framework\armi\operators\operatorMPI.py", line 149, in workerOperate
    self.workerQuit()
  File "...\home\jhader\natrium\framework\armi\operators\operatorMPI.py", line 230, in workerQuit
    runLog.close()  # no more messages.
  File "...\home\jhader\natrium\framework\armi\runLog.py", line 254, in close
    _ = [h.close() for h in LOG.stderrLogger.logger.handlers]
AttributeError: 'Logger' object has no attribute 'logger'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "...\home\jhader\natrium\nala\__main__.py", line 24, in main
    code = ArmiCLI().run()
  File "...\home\jhader\natrium\framework\armi\cli\__init__.py", line 189, in run
    return self.executeCommand(args.command, args.args)
  File "...\home\jhader\natrium\framework\armi\cli\__init__.py", line 218, in executeCommand
    return cmd.invoke()
  File "...\home\jhader\natrium\framework\armi\cli\run.py", line 33, in invoke
    inputCase.run()
  File "...\home\jhader\natrium\framework\armi\cases\case.py", line 348, in run
    o.operate()
  File "...\home\jhader\natrium\framework\armi\operators\operatorMPI.py", line 111, in operate
    traceback.print_exc()
  File "...\apps\dev\Python\3.7.2\lib\traceback.py", line 163, in print_exc
    print_exception(*sys.exc_info(), limit=limit, file=file, chain=chain)
  File "...\apps\dev\Python\3.7.2\lib\traceback.py", line 105, in print_exception
    print(line, file=file, end="")
AttributeError: 'Logger' object has no attribute 'write'
```

**Comment by theJollySin** [comment]
Darn and/or shucks.

Okay, one at a time:

1. This would be quite surprising to me. Especially since we are literally `print`ing the log messages at the end of the run: 
     https://github.com/terrapower/armi/blob/master/armi/runLog.py#L292
    I would give it a possibility that the stdlib python.logging.StreamOutput passes all text to stderr, instead of stdout. Though, that would be news to me. I can look.
2. That's not good!  Can you give me an example piece of code or test or something to run to reproduce this?  I certainly thought I tested for it. 
3. This is something I think I understand. And I will try to fix it. Though... something is odd there if you are seeing this and I certainly didn't. I even have a unit test that calls this method.

**Comment by jakehader** [comment]
Hi @john-science, yes I can point you to some outputs if you message me directly. For now I am going to revert this commit, test things out and push a separate remote branch to work off of until we work out the kinks.

**Comment by jakehader** [comment]
Update: After reverting this commit I am noticing that the head process information is actually set to `xtra` so maybe it's not a problem with writing out, but maybe the `xtra` verbosity isn't being treated correctly?

**Comment by john-science** [comment]
> Update: After reverting this commit I am noticing that the head process information is actually set to `xtra` so maybe it's not a problem with writing out, but maybe the `xtra` verbosity isn't being treated correctly?
> 
> Reverted branch - https://github.com/terrapower/armi/tree/revertLogger

IF we look back at the `master` branch from a few weeks ago, we can see that `xtra` was never something you called directly, it was something that was displayed. But `extra` was the thing we called:

https://github.com/terrapower/armi/blob/27358cfde190ab66605dc9c67b19d654c80fc3e9/armi/runLog.py#L53

Is this what you are talking about?

**Comment by jakehader** [comment]
Yeah that's what I am referring to

**Comment by jakehader** [comment]
Fixed with #367! Thanks for the help @john-science
